File: TypeScript/src/session2.td/agregation/Nat.ts
import {SemiAnneauUnitaireEuclidien } from '../../hierarchie/semianneauunitaireeuclidien';
import {FabriqueNaturels} from '../../session1.td/fabriquenaturels';
import {EtatNaturel} from './etatnaturel';
import {EtatNaturelPur} from "./EtatNaturelPur";

export interface Nat extends FabriqueNaturels<Nat>, SemiAnneauUnitaireEuclidien<Nat>, EtatNaturel<Nat> {

	creerNatAvecEtat(e : EtatNaturelPur) : Nat;
	etat() : EtatNaturelPur;

	equals(o : Nat) : boolean;
}

File: Java/src/session2/td/agregation/Nat.java
package session2.td.agregation;

import hierarchie.SemiAnneauUnitaireEuclidien;
import session1.td.FabriqueNaturels;

public interface Nat extends FabriqueNaturels<Nat>, SemiAnneauUnitaireEuclidien<Nat>, EtatNaturel<Nat> {

	Nat creerNatAvecEtat(EtatNaturelPur e);
	EtatNaturelPur etat() ;

}
